### PR TITLE
feat: add build command

### PR DESCRIPTION
**Pull request by Kamandlou**
Closes [#359](https://github.com/goravel/goravel/issues/359)

## 📑 Description
add build command

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

**Comment by Kamandlou** [comment]
Hi @hwbrzzl,
This PR is Review Ready.

File: console/console/build_command.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
+package console
+
+import (
+	"fmt"
+	"github.com/gookit/color"
+	"github.com/goravel/framework/contracts/config"
+	"github.com/goravel/framework/contracts/console"
+	"github.com/goravel/framework/contracts/console/command"
+	"os"
+	"os/exec"
```
Comment (hwbrzzl) [inline console/console/build_command.go:10]: Please sort import

```suggestion
	"fmt"
	"os"
	"os/exec"
	
	"github.com/gookit/color"
	
	"github.com/goravel/framework/contracts/config"
	"github.com/goravel/framework/contracts/console"
	"github.com/goravel/framework/contracts/console/command"
```

File: console/console/build_command.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+)
+
+type BuildCommand struct {
+	config config.Config
+}
+
+func NewBuildCommand(config config.Config) *BuildCommand {
```
Comment (hwbrzzl) [inline console/console/build_command.go:17]: Need register this command to `console/service_provider.go`.

File: console/console/build_command.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+	return "Build the application"
+}
+
+// Extend The console command extend.
+func (receiver *BuildCommand) Extend() command.Extend {
+	return command.Extend{
+		Category: "build",
+		Flags: []command.Flag{
+			&command.StringFlag{
+				Name:  "system",
```
Comment (hwbrzzl) [inline console/console/build_command.go:39]:
```suggestion
				Name:  "system",
				Aliases: []string{"s"},
```

File: console/console/build_command.go [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+		}
+
+		if result != "yes" {
+			color.Yellowln("Command Canceled")
+
+			return nil
+		}
+	}
+
+	system := ctx.Option("system")
```
Comment (hwbrzzl) [inline console/console/build_command.go:70]: Need to judge the value, should be: `linux`, `windows`, or `darwin`.

File: console/console/build_command_test.go [hunk 5]
```diff
@@ -0,0 +1,54 @@
+package console
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	configmock "github.com/goravel/framework/mocks/config"
+	consolemocks "github.com/goravel/framework/mocks/console"
```
Comment (hwbrzzl) [inline console/console/build_command_test.go:10]:
```suggestion
	mocksconfig "github.com/goravel/framework/mocks/config"
	mocksconsole "github.com/goravel/framework/mocks/console"
```

**Review by hwbrzzl — COMMENTED** [review]
A beautiful PR. 👍

**Comment by Kamandlou** [comment]
@hwbrzzl

File: console/console/build_command.go [hunk 6]
```diff
@@ -0,0 +1,103 @@
+package console
+
+import (
+	"fmt"
+	"golang.org/x/exp/slices"
```
Comment (hwbrzzl) [inline console/console/build_command.go:5]: The `exp` package is unstable, we should not use it, please use https://pkg.go.dev/slices instead.

File: console/console/build_command.go [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+		if result != "yes" {
+			color.Yellowln("Command Canceled")
+
+			return nil
+		}
+	}
+
+	system := ctx.Option("system")
+	if !slices.Contains([]string{"linux", "windows", "darwin"}, system) {
+		color.Redln("The value of the --system flag is invalid.")
```
Comment (hwbrzzl) [inline console/console/build_command.go:76]: Please list the valid value here.

File: console/console/build_command.go [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+		if result != "yes" {
+			color.Yellowln("Command Canceled")
+
+			return nil
+		}
+	}
+
+	system := ctx.Option("system")
+	if !slices.Contains([]string{"linux", "windows", "darwin"}, system) {
```
Comment (hwbrzzl) [inline console/console/build_command.go:75]: Please add test case for this flow.

**Comment by hwbrzzl** [comment]
CI failed

**Review by hwbrzzl — APPROVED** [review]
Perfect 👍

**Pull request merged**